### PR TITLE
Handle custom stream

### PR DESCRIPTION
**Pull request by wxt2005**
Recently I run into a problem while using Request with form-data, I tried to submit a custom stream provided by another library, but then I found form-data incorrectly calculated the content length.
I searched for issues and PRs, then I found #70, for unfortunately it's not be merged. So I made this new PR, all codes except minor changes are come from #70.

### CODE REVIEW

**Comment by alexindigo** [comment]
Thank you for the pull-request. Do you mind to update Readme with new state of things?

I restarted broken windows build, seems like it was an environment issue.

**Comment by wxt2005** [comment]
Of course. I've added notes about those new behaviors.

**Comment by giautm** [comment]
I ran into the same issues when I trying forward incoming stream from Multer to the storage service. I have used the fork of @wxt2005 and its working as perfect. 👍Pleaseeee, merge this soon. Tks. :)

```js
var crypto = require('crypto');
var FormData = require('form-data');
var fetch = require('node-fetch');

function getFilename (req, file, cb) {
  crypto.pseudoRandomBytes(16, function (err, raw) {
    cb(err, err ? undefined : raw.toString('hex'));
  });
}

function getBucketTmp (req, file, cb) {
  cb(null, 'tmp');
}

function TalariaStorage(opts) {
  this.serviceUrl = opts.serviceUrl;
  if (!this.serviceUrl) {
    throw new Error('Missing `serviceUrl` options');
  }

  this.getFilename = (opts.filename || getFilename);
  if (typeof opts.bucket === 'string') {
    this.getBucket = ($0, $1, cb) => cb(null, opts.bucket);
  } else {
    this.getBucket = (opts.bucket || getBucketTmp);
  }
}

TalariaStorage.prototype._handleFile = function _handleFile (req, file, cb) {
  this.getBucket(req, file, (err, bucket) => {
    if (err) return cb(err);

    this.getFilename(req, file, (err, filename) => {
      if (err) return cb(err);

      var form = new FormData();
      form.append('bucket', bucket);
      form.append('file', file.stream, filename);

      fetch(this.serviceUrl, {
        method: 'POST',
        body: form,
      }).then(res => res.json())
        .then(result => cb(null, {
          bucket,
          filename,
          key: result.key,
          location: result.url,
        }))
        .catch(cb);
    });
  });
};

TalariaStorage.prototype._removeFile = function _removeFile (req, file, cb) {
  cb(null);
};

module.exports = function (opts) {
  return new TalariaStorage(opts);
};
```

**Comment by anthnyprschka** [comment]
I had a similar problem and @wxt2005's fork fixed it for me. Then I discovered that request allows you to pass a `knownLength` parameter to the options of a formData property. This also fixes the problem for custom streams whose length you know beforehand

File: Readme.md [hunk 1]
```diff
@@ -227,6 +227,8 @@ fetch('http://example.com', { method: 'POST', body: form })
 ## Notes
 
 - ```getLengthSync()``` method DOESN'T calculate length for streams, use ```knownLength``` options as workaround.
+- ```getLength(cb)``` will send an error as first parameter of callback if stream length cannot be calculated (e.g. send in custom streams w/o using ```knownLength```).
+- ```sbumit``` will not add `content-length` if form length is unknown or not calculable.
```
Comment (TPXP) [inline Readme.md]: There's a typo here, you probably meant `submit`

**Comment by thepheer** [comment]
Any chance of merging this in near future?

**Comment by psychowico** [comment]
I have same issue. do we wait 5 months just for fixing typo in README?
can we merge it already?

**Comment by wxt2005** [comment]
Hi,@TPXP, sorry I haven't noticed your comment before, now the typo has been fixed.

**Comment by nytamin** [comment]
I have had the same issue, can this be merged please?

**Comment by niftylettuce** [comment]
I'm now a maintainer.  v4.0.0 released https://github.com/form-data/form-data/releases/tag/v4.0.0

**Comment by gish** [comment]
Just out of curiosity, what warranted a major version bump? I would have assumed this bugfix should have been a patch bump.

**Pull request merged**